src/Microsoft.Sbom.Api/Executors/Spdx2SerializationStrategy.cs

```diff
@@ -0,0 +1,162 @@
+        relationshipsArrayGenerator.SbomConfig = sbomConfig;
+        externalDocumentReferenceGenerator.SbomConfig = sbomConfig;
+
+        fileArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        packageArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        relationshipsArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        externalDocumentReferenceGenerator.SpdxManifestVersion = spdxManifestVersion;
+
+        var errors = new List<FileValidationResult>();
+
+        await WriteFiles(fileArrayGenerator);
+
+        await WritePackages(packageArrayGenerator);
+
+        await WriteExternalDocRefs(externalDocumentReferenceGenerator);
+
+        await WriteRelationships(relationshipsArrayGenerator);
+
+        return errors;
+    }
+
+    /// <summary>
+    /// Write to Files section
+    /// </summary>
+    /// <param name="fileArrayGenerator"></param>
+    /// <returns></returns>
+    private async Task WriteFiles(IJsonArrayGenerator<FileArrayGenerator> fileArrayGenerator)
+    {
+        var filesGenerateResult = await fileArrayGenerator.GenerateAsync();
+        filesGenerateResult.Errors.AddRange(filesGenerateResult.Errors);
+        WriteJsonObjectsFromGenerationResult(filesGenerateResult, fileArrayGenerator.SbomConfig);
+    }
+
+    /// <summary>
+    /// Write to Packages section
+    /// </summary>
+    /// <param name="packageArrayGenerator"></param>
+    /// <returns></returns>
+    private async Task WritePackages(IJsonArrayGenerator<PackageArrayGenerator> packageArrayGenerator)
+    {
+        var packagesGenerateResult = await packageArrayGenerator.GenerateAsync();
+        packagesGenerateResult.Errors.AddRange(packagesGenerateResult.Errors);
+        WriteJsonObjectsFromGenerationResult(packagesGenerateResult, packageArrayGenerator.SbomConfig);
+    }
+
+    /// <summary>
+    /// Write to External Document Reference section
+    /// </summary>
+    /// <param name="externalDocumentReferenceGenerator"></param>
+    /// <returns></returns>
+    private async Task WriteExternalDocRefs(IJsonArrayGenerator<ExternalDocumentReferenceGenerator> externalDocumentReferenceGenerator)
+    {
+        var externalDocumentReferenceGenerateResult = await externalDocumentReferenceGenerator.GenerateAsync();
+        externalDocumentReferenceGenerateResult.Errors.AddRange(externalDocumentReferenceGenerateResult.Errors);
+        WriteJsonObjectsFromGenerationResult(externalDocumentReferenceGenerateResult, externalDocumentReferenceGenerator.SbomConfig, externalDocumentReferenceGenerateResult.SourcesProviders);
+    }
+
+    /// <summary>
+    /// Write to Relationships section
+    /// </summary>
+    /// <param name="relationshipsArrayGenerator"></param>
+    /// <returns></returns>
+    private async Task WriteRelationships(IJsonArrayGenerator<RelationshipsArrayGenerator> relationshipsArrayGenerator)
+    {
+        var relationshipGenerateResult = await relationshipsArrayGenerator.GenerateAsync();
```

Comment:
Variable name didn't change when `GenerateResult` type was renamed to `GenerationResult`